FlagsmithClient/src/test/java/com/flagsmith/RealTimeUpdatesIntegrationTests.kt

```diff
@@ -0,0 +1,216 @@
+
+    @After
+    fun tearDown() {
+    }
+
+    private fun setupMocks() {
+        MockitoAnnotations.initMocks(this)
+
+        Mockito.`when`(mockApplicationContext.getResources()).thenReturn(mockContextResources)
+        Mockito.`when`(
+            mockApplicationContext.getSharedPreferences(
+                ArgumentMatchers.anyString(),
+                ArgumentMatchers.anyInt()
+            )
+        ).thenReturn(
+            mockSharedPreferences
+        )
+        Mockito.`when`(mockApplicationContext.cacheDir).thenReturn(File("cache"))
+
+        Mockito.`when`(mockContextResources.getString(ArgumentMatchers.anyInt())).thenReturn("mocked string")
+        Mockito.`when`(mockContextResources.getStringArray(ArgumentMatchers.anyInt())).thenReturn(
+            arrayOf(
+                "mocked string 1",
+                "mocked string 2"
+            )
+        )
+        Mockito.`when`(mockContextResources.getColor(ArgumentMatchers.anyInt())).thenReturn(Color.BLACK)
+        Mockito.`when`(mockContextResources.getBoolean(ArgumentMatchers.anyInt())).thenReturn(false)
+        Mockito.`when`(mockContextResources.getDimension(ArgumentMatchers.anyInt())).thenReturn(100f)
+        Mockito.`when`(mockContextResources.getIntArray(ArgumentMatchers.anyInt()))
+            .thenReturn(intArrayOf(1, 2, 3))
+    }
+
+    @Test
+    fun testEnvironmentVariablesArentEmpty() {
+        Assert.assertTrue(environmentKey.isNotEmpty())
+        Assert.assertTrue(apiToken.isNotEmpty())
+        Assert.assertTrue(featureId.isNotEmpty())
+        Assert.assertTrue(featureStateId.isNotEmpty())
+    }
+
+    /// Update after 5 secs, should be done in 60 seconds or fail
+    @Test(timeout = 60000)
+    fun testGettingFlagsWithRealtimeUpdatesAfterPuttingNewValue() = runBlocking {
+        // Get the current value
+        val currentFlagValueString =
+            flagsmith.getValueForFeatureSync(featureId).getOrThrow() as String?
+        println("Type of currentFlagValueDouble: ${currentFlagValueString?.javaClass?.name}")
+        Assert.assertNotNull(currentFlagValueString)
+        val currentFlagValue: String = currentFlagValueString!!
+
+        // After 5 seconds try to update the value using the retrofit service
+        CoroutineScope(Dispatchers.IO).launch {
+            // Wait 5 seconds before updating the value
+            delay(5000)
+
+            val response = retrofitService
+                .setFeatureStates(authToken, featureStateId, environmentKey!!, FeatureStatePutBody(true, "new-value"))
+                .execute()
+            if (!response.isSuccessful) println("ERROR response: $response")
+            println("Response: $response")
+            Assert.assertTrue(response.isSuccessful)
+        }
+
+        var newUpdatedFeatureValue: String? = ""
+        do {
+            newUpdatedFeatureValue = flagsmith.flagUpdateFlow.value
+                .find { flag -> flag.feature.name == featureId }?.featureStateValue as String? ?: ""
```

Comment:
Because code on mobile doesn't always live for very long things like listeners have to be tied to the lifecycle of the class that uses it so that they're properly destructed etc. This is the standard way to do this.